audformat/core/utils.py

```diff
@@ -640,6 +641,62 @@ def same_dtype(d1, d2) -> bool:
     return d1.name == d2.name
 
 
+def set_file_extension(
+        index: pd.Index,
+        extension: str,
+        pattern: str = r'\.[a-zA-Z0-9]+$',
+) -> pd.Index:
+    r"""Change the file extension of index entries.
+
+    It replaces all existing file extensions
+    in the index file path
+    by the new provided one.
+
+    Args:
+        index: index with file path
+            conform to :ref:`table specifications <data-tables:Tables>`
+        extension: new file extension without ``'.'``.
+            If set to ``''``,
+            the current file extension is removed
+        pattern: regexp pattern to match current extensions.
+            In contrast to ``extension``,
+            you have to include ``'.'``
+
+    Returns:
+        updated index
+
+    Example:
+        >>> idx = filewise_index(['f1.wav', 'f2.flac'])
```

Comment:
For consistency with the other examples I would rename to `index`,

Reply:
There is not really consitency ;)

![image](https://user-images.githubusercontent.com/173624/142168700-6efb3fe5-8ff0-4918-b59c-0af666b1ce83.png)

![image](https://user-images.githubusercontent.com/173624/142168753-8071443f-658d-4779-9041-eb3fe91233af.png)

![image](https://user-images.githubusercontent.com/173624/142168862-01e32448-a65c-459d-9a59-5a8109d56c4a.png)

![image](https://user-images.githubusercontent.com/173624/142168967-46c9e2ea-a3e7-4d21-8977-82f2ece3a82f.png)

But I agree, that `index` makes the most sense here. I changed it accordingly

Reply:
Agree, `i1` and `i2` is really bad naming. I would like to say it wasn't me who added those examples, but I'm afraid that'd be a lie :)